File: pulse-kotlin/src/commonTest/kotlin/kotlin/com/fluxchess/pulse/kotlin/PieceTypeTest.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package kotlin.com.fluxchess.pulse.kotlin

import com.fluxchess.pulse.kotlin.pieceTypes
import kotlin.test.Test
import kotlin.test.assertEquals

class PieceTypeTest {
	@Test
	fun `PieceTypes array should be valid`() {
		for (pieceType in pieceTypes.indices) {
			assertEquals(pieceTypes[pieceType], pieceType)
		}
	}
}

File: pulse-go/internal/pulse/piece_type_test.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package pulse

import (
	"testing"
)

func TestPieceType(t *testing.T) {
	t.Run("PieceTypes array should be valid", func(t *testing.T) {
		for pieceType := range PieceTypes {
			if PieceTypes[pieceType] != pieceType {
				t.Errorf("wanted piece types at index %d to be %v, got %v", pieceType, pieceType, PieceTypes[pieceType])
			}
		}
	})
}
